### ISSUE TITLE
播放长视频时.分割的视频之间会跳帧

### ISSUE THREAD

**Issue by CenJing** [typcn/bilibili-mac-client#234]
播放长视频时.分割的视频之间会跳帧
比如一部视频40分钟.他每10分钟分割一个文件.
播放到10:00的时候.会突然跳到10:10.且后退也无法播放跳过的10秒.拖动进度条也是不行.测试在早期2.26的版本已经出现.更早版本未测试.
求修复.

**Comment by typcn** [comment]
给一个样例地址？

**Comment by CenJing** [comment]
比如我现在看的
http://www.bilibili.com/video/av3780502/index_5.html
30:05会跳到30:14

**Comment by typcn** [comment]
等有时间我去 mpv 那边反馈下。。

**Comment by CenJing** [comment]
现在有办法解决不?这个BUG导致看很多片子很麻烦.

**Comment by typcn** [comment]
可以在设置中开启 MP4 模式，画质会降一点，但是只有一个分段

**Comment by CenJing** [comment]
我之前一直是开启MP4的.但也是会出现这个情况

**Comment by typcn** [comment]
B 站对 mp4 不会分段的啊。。

**Comment by CenJing** [comment]
你尝试开启那个MP4模式.然后再播放我刚刚的视频.看看30:05

**Comment by typcn** [comment]
这是压的问题吧。。

**Comment by CenJing** [comment]
啊?很多视频也出现这种问题.

**Comment by cnbeining** [comment]
This is probably due to gaps between I frames are tooooo long. 

Although longer time lap would help reduce the file size, it should also be noticed that seeking would be screwed with that. Should you are using other players(e.g., mpv), you can notice mosaic on the graph.

TL;DR:  Not something we can do from our side.

**Comment by CenJing** [comment]
http://www.bilibili.com/video/av4064398/ 6:04->6:14

**Comment by cnbeining** [comment]
Are you sure you are not jumping between different slices of the video? This problem can be not avoidable.

**Comment by typcn** [comment]
我回去把关键帧拆开看下

**Comment by typcn** [comment]
@cnbeining That's very hard to avoid on client side , but is easy to do that on server , they are using very bad slice method....

**Comment by cnbeining** [comment]
I doubt whether they will fix their problem.

Not gonna do.

**Comment by typcn** [comment]
It should be fixed ....

You can test it by clone this repo and build with Xcode.

I will publish update tomorrow

**Comment by typcn** [comment]
Click "Check for update" to get new version

**Comment by CenJing** [comment]
Thank you very much.
